src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+export enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
+
+  constructor(
+    public color: Color,
+
+    public a: number,
+
+    public b: number,
+
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
```

Comment:
You don't need to use `this` inside of the constructor. It is redundant. The same in other constructors.
```suggestion
    if (a <= 0 || b <= 0 || c <= 0) {
```